client/app/services/designer-state.service.js

```diff
@@ -141,12 +105,19 @@
 
       function matchAtomicServiceItemToTabs(srvTemplate) {
         var subTab;
-        var newItem;
         if (srvTemplate.service_type === 'atomic') {
           if (srvTemplate.prov_type !== 'generic') {
             subTab = findSubTabByProvType(srvTemplate.prov_type);
-          } else {
+            if (!subTab) {
+              console.log("Couldn't find toolbox Tab for Prov_Type: " + srvTemplate.prov_type);
```

Comment:
@dtaylor113 I appreciate the position you are in, but I agree with @AllenBW console.log isn't the best choice, and I'm weary of "I'll fix it later" type comments, because sometimes one just doesn't get an opportunity to fix.